RecoLocalCalo/HcalRecAlgos/plugins/HcalRecAlgoESProducer.cc

```diff
@@ -94,5 +96,19 @@ HcalRecAlgoESProducer::produce(const HcalSeverityLevelComputerRcd& iRecord)
    return myComputer ;
 }
 
+void HcalRecAlgoESProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<uint32_t>("phase",0);
+    desc.add<std::vector<std::string>>("RecoveredRecHitBits");
+    desc.add<std::vector<std::string>>("DropChannelStatusBits");
+    descriptions.add("hcalRecAlgos",desc);
+
+    edm::ParameterSetDescription desc_sevlvl;
+    desc_sevlvl.add<uint32_t>("Level");
+    desc_sevlvl.add<std::vector<std::string>>("RecHitFlags");
+    desc_sevlvl.add<std::vector<std::string>>("ChannelStatus");
+    desc_sevlvl.addVPSet("SeverityLevels",desc_sevlvl);
+}
```

Comment:
@Martin-Grunewald 

Now the branch is updated as follows.

* The `fillDescriptions` has the full definition of parameters. I used the `edmPythonConfigToCppValidation` to generate it.
* The explicit cfi file `RecoLocalCalo/HcalRecAlgos/python/hcalRecAlgoESProd_cfi.py` uses the generated cfi file and has only era customization

After the update, I confirmed that 

* `runTheMatrix.py -l 10024.0 --useInput all &` test passed without failure
* The DQM plots looked at before (https://github.com/cms-sw/cmssw/pull/19532#issuecomment-315703078) are same
* The eta of calojet is same